IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework.Tests/TestBuilderTests/PositiveConditions/LoggerTests.cs

```diff
@@ -0,0 +1,188 @@
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void RemovedLoggerDoesNotThrowWhenAttemptingToActivateExecuteArg()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockExecuteArg>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void CustomLoggerAddsWithoutError()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .AddLogger<CustomLogger>()
+                .AddTestBlock<CustomLogBlock>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void RemovingLoggerTwiceDoesNotThrow()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .RemoveLogger()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockProp>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void AddingLoggerThanRemovingDoesNotThrow()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .AddLogger<CustomLogger>()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockProp>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+    }
+
+    public class DefaultLogBlock : TestBlock
+    {
+        public void Execute()
+        {
+            Assert.NotNull(Log);
+            Assert.Equal(typeof(DebugLogger), Log!.GetType());
+        }
+    }
+
+    public class RemovedLogBlockProp : TestBlock
+    {
+        public void Execute()
+        {
+            Assert.Null(Log);
+        }
+    }
+
+    public class RemovedLogBlockCtor : TestBlock
+    {
+        public RemovedLogBlockCtor(ITestCaseLogger? log)
+        {
+            Log = log;
+        }
+
+        public void Execute()
+        {
+            Assert.Null(Log);
+        }
+    }
+
+    public class RemovedLogBlockExecuteArg : TestBlock
+    {
+        public void Execute(ITestCaseLogger? log)
+        {
+            Assert.Null(log);
+        }
+    }
+
+    public class CustomLogBlock : TestBlock
+    {
+        public void Execute()
+        {
+            Assert.NotNull(Log);
+            Assert.Equal(typeof(CustomLogger), Log!.GetType());
+            if(Log is CustomLogger cl)
+            {
+                Assert.True(cl.Invocations > 0);
+            }
+        }
+    }
+
+    public class CustomLogger : ITestCaseLogger
+    {
+        public CustomLogger(TestCase tc)
+        {
+            TestCase = tc;
+        }
+        public TestCase TestCase { get; }
+
+        public string? CurrentTestBlock { get; set; }
+
+        public int Invocations { get; set; }
```

Comment:
It looks like this is only compared against `> 0`. Typically if that is all it is good for, it would  better to just make this a boolean.